code/main/edge/edgeRequestWorker.brs

```diff
@@ -87,72 +89,172 @@ function _adb_EdgeRequestWorker(edgeResponseManager as object) as object
         '
         ' @return an array of EdgeResponse objects
         ' ------------------------------------------------------------------------------------------
-        processRequests: function(configId as string, ecid as string, edgeDomain = invalid as dynamic) as dynamic
+        processRequests: function(edgeConfig as object) as dynamic
             responseArray = []
             while m.hasQueuedEvent()
 
-                currTS = _adb_timestampInMillis()
-                if (m._lastFailedRequestTS <> m._INVALID_WAIT_TIME) and ((currTS - m._lastFailedRequestTS) < m._RETRY_WAIT_TIME_MS)
-                    ' Wait for 30 seconds before retrying the hit failed with recoverable error.
+                if m._shouldWaitBeforeRetry()
+                    ' Wait for 30 seconds before retrying the request
                     exit while
                 end if
 
                 ' grab oldest hit in the queue
-                requestEntity = m._queue.Shift()
+                requestToBeSent = invalid
+
+                if m._isBlockedByConsent(m._consentState)
+                    ' since the consent is pending process the consent request
+                    consentRequest = m._consentQueue.Shift()
+                    if not _adb_isEdgeConsentRequest(consentRequest)
```

Comment:
This section will be more readable if you just check 
```
if consentQueue is empty 
	exit processing
```

Reply:
okay